### PR TITLE
Initial Changes for GEMS + SPATIAL

### PR DESCRIPTION
**Pull request by RadhaGulhane13**

### CODE REVIEW

File: src/torchgems/train_spatial.py [hunk 1]
```diff
@@ -22,6 +22,46 @@
 import torch.distributed as dist
 
 
+def isPowerTwo(num):
```
Comment (Quentin-Anthony) [inline src/torchgems/train_spatial.py:25]: put this in a utils file instead of redefining

File: src/torchgems/train_spatial_master.py [hunk 2]
```diff
@@ -17,11 +17,77 @@
 # limitations under the License.
 
 
-from torchgems.train_spatial import train_model_spatial
+from torchgems.train_spatial import train_model_spatial, verify_spatial_config
 import torch
 import torch.distributed as dist
 
 
+def isPowerTwo(num):
```
Comment (Quentin-Anthony) [inline src/torchgems/train_spatial_master.py:25]: utils file

File: benchmarks/gems_master_with_spatial_parallelism/benchmark_amoebanet_gems_master_with_sp.py [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+
+    def writelines(self, datas):
+        self.stream.writelines(datas)
+        self.stream.flush()
+
+    def __getattr__(self, attr):
+        return getattr(self.stream, attr)
+
+
+def init_processes(backend="tcp"):
```
Comment (Quentin-Anthony) [inline benchmarks/gems_master_with_spatial_parallelism/benchmark_amoebanet_gems_master_with_sp.py:69]: this default should always be `mpi`. Change https://github.com/OSU-Nowlab/MPI4DL/blob/c6716708b11501fe6a6b995b39152d6d0e636176/benchmarks/spatial_parallelism/benchmark_amoebanet_sp.py#L65 as well

File: benchmarks/gems_master_with_spatial_parallelism/benchmark_amoebanet_gems_master_with_sp.py [hunk 4]
```diff
@@ -0,0 +1,485 @@
...
+    """Initialize the distributed environment."""
+    dist.init_process_group(backend)
+    size = dist.get_world_size()
+    rank = dist.get_rank()
+    return size, rank
+
+
+def get_depth(version, n):
```
Comment (Quentin-Anthony) [inline benchmarks/gems_master_with_spatial_parallelism/benchmark_amoebanet_gems_master_with_sp.py:77]: Should also be a util, and should be documented

File: src/torchgems/train_spatial.py (hunk 1 above)
Comment (RadhaGulhane13) [reply src/torchgems/train_spatial.py]: Added this in utils.py

**Review by Quentin-Anthony — APPROVED** [review]

**Pull request merged**